lib/app_profiler/base_profile.rb

```diff
@@ -17,15 +16,15 @@ class << self
       def from_stackprof(data)
         options = INTERNAL_METADATA_KEYS.map { |key| [key, data[:metadata]&.delete(key)] }.to_h
 
-        StackprofProfile.new(data, **options).tap do |profile|
+        StackprofProfile.new(data, id: ProfileId.current, **options).tap do |profile|
           raise ArgumentError, "invalid profile data" unless profile.valid?
         end
       end
 
       def from_vernier(data)
         options = INTERNAL_METADATA_KEYS.map { |key| [key, data[:meta]&.delete(key)] }.to_h
 
-        VernierProfile.new(data, **options).tap do |profile|
+        VernierProfile.new(data, id: ProfileId.current, **options).tap do |profile|
```

Comment:
The profile ID would be wrong if the ID option is passed in. If the ID option is passed in, we should probably assign the ID, right?

Reply:
you are right -- i guess we would like to keep the existing behaviour intact. I have changed this -- so when its not supplied we will fallback

`@id  = id.presence || ProfileId.current`